### PR TITLE
Validationstesting

### PR DESCRIPTION
**Pull request by Wyattwicks**
Necessary checkmarks:
    [x] All Tests are Passing
    [x] The code will run locally

Type of change:
    [] New feature
    [] Bug Fix
    [x] Improvement / Refactor

Any Requests From Reviewer:

Testing Changes:
    [x] Added new tests
    [] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:
    [x] My code has no unused/commented out code
    [x] I have reviewed my code
    [x] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(As Needed)Please Include screenshot/screencast of new feature ect.

### CODE REVIEW

**Comment by Wyattwicks** [comment]
NEEDS TO BE REBASED AND MERGED BEFORE MERGE!!!!!

**Review by sarahlane8 — COMMENTED** [review]
this is so interesting! the format is pretty similar to front end testing, just slightly different syntax, but the same with it blocks, executions, and expect statements! can't wait to review your code more closely when this is done!

File: spec/requests/api/v1/sessions_spec.rb [hunk 1]
```diff
@@ -37,4 +37,26 @@
...
+
+      VCR.use_cassette('user_login') do
+        post '/api/v1/sessions', headers: headers, params: user_params.to_json
+      end
+
+      user = JSON.parse(response.body, symbolize_names: true)
+      expect(response.status).to eq(400)
+      expect(user).to eq({:error=>"credentials are bad"})
+    end
+  end
```
Comment (Thee-Dust) [inline spec/requests/api/v1/sessions_spec.rb:61]: i like how the sad path is its own test

Comment (harrison-blake) [reply spec/requests/api/v1/sessions_spec.rb:61]: LOOKS GOOD

**Review by harrison-blake — APPROVED** [review]

**Pull request merged**